mozci/push.py

```diff
@@ -287,8 +288,19 @@ def tasks(self):
             ],
             return_when=concurrent.futures.FIRST_EXCEPTION,
         )
+        failed = set()
         for f in done:
-            f.result()
+            try:
+                f.result()
+            except SourcesNotFound as e:
```

Comment:
Instead of ignoring these errors, couldn't we avoid trying to retrieve groups for such tasks?

Reply:
I guess this is what your next commit is doing at https://github.com/mozilla/mozci/pull/455/commits/92dc36226900dd8115a0802e6ae9e2d3dd61b51a#diff-cbdcefec12b605b952245761dbfbbbb2ec7fc73addcc510c2b8dc48a60bd5d02R288? Why is this still necessary then?

Reply:
There are still edge cases that hit this, even with the next patch. I can't remember off-hand what task it was.. I guess it was missing from the "no groups suite" list. Either way, I think we want this guard since we don't want to regress `mozci` every time someone adds a new suite that doesn't have groups. Especially when groups might not even be needed for the specific use case (like in Sarah's case).

Reply:
The data source could just return an empty list when the task doesn't have groups

Reply:
In the treeherder source's case we purposefully [raise ContractNotFilled](https://github.com/mozilla/mozci/blob/master/mozci/data/sources/treeherder/__init__.py#L44) so that we fallback to the artifact data source on older pushes that are no longer in the treeherder DB.

We could make the artifact data source return an empty list, but then what if you don't have it configured in your `config.toml`? Am I missing a downside to catching and warning here?

Reply:
The only downsides are the additional (even if limited) complexity of the logic and the possibility of ignoring some real errors (e.g. say treeherder suddenly stops returning results for all tasks, we won't notice until we look at the logs).
They're relatively limited downsides though, so feel free to land this as-is :)

Reply:
I think the root issue with your suggestion is that there's no proper way of distinguishing between the "push isn't in the DB" and "the task has no groups" in the Treeherder API. I guess we could check if there exists some result that gets returned, but that method of distinguishing would fail when e.g there is a build only push.